### PR TITLE
test: Add MTL edge-case tests, checker parity tests, and migrate to mise

### PR DESCRIPTION
**Pull request by rizumita**
## Summary
- Add 40 MTL timed operator edge-case tests covering boundary precision, zero-width intervals, duplicate timestamps, vacuous intervals, and startIndex shifting
- Add 324 core/MTL evaluator parity matrix tests (6 traces × 15 formulas × multiple startIndex + nested formulas + Release/WeakUntil variants)
- Add 8 StreamLtlChecker vs StreamMtlChecker parity tests documenting semantic divergence (last-index vs index-0 evaluation)
- Migrate Flutter version management from fvm (.fvmrc) to mise (mise.toml)

## Test plan
- [x] `dart test packages/temporal_logic_mtl/test/mtl_operators_test.dart` — 40 passed
- [x] `dart test packages/temporal_logic_mtl/test/semantics_matrix_test.dart` — 324 passed
- [x] `flutter test test/checker_parity_test.dart` (in flutter package) — 8 passed
- [x] All existing tests pass: core 94 (~1 skip), MTL 412, Flutter 69

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## リリースノート

* **Chores**
  * Flutter バージョン管理ツールを最新化しました。

* **Tests**
  * LTL/MTL チェッカーの動作検証テストを追加しました。
  * MTL オペレータの評価テストを追加しました。
  * 複数評価エンジン間の互換性検証テストを追加しました。

### CODE REVIEW

File: mise.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[tools]
+flutter = "latest"
```
[the remark this answers is not in the document]
Comment (rizumita) [reply mise.toml]: 修正しました。 ✅

Flutter を `3.41.5` に固定し、`README.md` と `scripts/run_all_tests.sh` も `mise` 前提に更新しました。あわせて 3.41.5 で再解決された lockfile を反映し、互換性のため `examples/snackbar_mtl` の `fake_async` 制約も更新しています。

`bash scripts/run_all_tests.sh` でフルスイート通過を確認しました。

commit: 2219355

**Review by rizumita — COMMENTED** [review]
CodeRabbit review follow-up:

- `packages/temporal_logic_mtl/test/mtl_operators_test.dart`: removed the duplicated `Boundary precision` group in commit `2219355`.
- `packages/temporal_logic_flutter/test/checker_parity_test.dart`: kept `pFalse_f` as-is. I verified the file under Flutter `3.41.5`; it is still used by the `Until(pFalse_f, pTrue_f)` case, and removing it causes the test to fail.
- `mise.toml` / tooling docs: pinned Flutter to `3.41.5`, updated the repo docs/script to use `mise`, refreshed lockfiles, and verified with `bash scripts/run_all_tests.sh`.
